plugin.py

```diff
@@ -1213,22 +1176,21 @@ def checkout_pg_distant(self):
         self.versioning.checkout(tables_for_conninfo, user_selected_features)
 
         # add layers from offline version
-        grp_idx = self.iface.legendInterface().addGroup(working_copy_schema)
-        for layer_id in reversed(self.current_layers):
-            layer = QgsMapLayerRegistry.instance().mapLayer(layer_id)
-            new_uri = QgsDataSourceURI(layer.source())
+        layers = []
+        for layer in reversed(self.current_layers):
+            new_uri = QgsDataSourceUri(layer.source())
             new_uri.setDataSource(working_copy_schema,
                                   new_uri.table()+"_view",
                                   'geom',
                                   new_uri.sql(),
                                   new_uri.keyColumn())
             # TODO: IT'S UGLY
```

Comment:
Are there no more elegant solutions? or, please, remove my comments

Reply:
You mean the new_group method that take a layers with tuples. I'm ok this is not the best way to do but it avoid code copies and I didn't find better ways.